src/compiler.ts

```diff
@@ -7915,7 +7915,9 @@ export class Compiler extends DiagnosticEmitter {
           return module.unreachable();
         }
         let globalType = global.type;
-        assert(globalType != Type.void);
+        if(globalType == Type.void) {
```

Comment:
Missing a whitespace here, and wondering if the fix should instead make `compileGlobal` above return `false`. Might have unforeseen consequences.

Reply:
It shouldn't compile regardless- since a global resolving into a void is already an error:
```typescript
if (resolvedType == Type.void) {
          this.error(
            DiagnosticCode.Type_expected,
            typeNode.range
          );
          return false;
        }
```
Given that, this is really just a question of what gets displayed when it does fail. I suppose the worst case here is that there is some edge case where this will append additional errors which don't make sense to the error which the dev needs to see, but that doesn't seem like huge problem.

Reply:
I agree that this requires a fix, just questioning the chosen location. If `compileGlobal` would return `false` if it failed to compile that same global earlier already, which gives us an `unreachable` as well, this would be much clearer than checking for `Type.void`.

Reply:
compileGlobal does in fact return false for the first statement- and that false was being ignored by compileTopLevelStatement. If I move the unreachable call there it will also work:
```typescript
if (!this.compileGlobal(<Global>element)) this.module.unreachable();
```
Most other calls to compileGlobal do the same, checking its result and then returning unreachable if it failed. Is this what you mean?
 
For this to work however I'd still need to delete the globalType != void assertion, otherwise none of these errors will ever print.

Reply:
> Most other calls to compileGlobal do the same, checking its result and then returning unreachable if it failed. Is this what you mean?

Yeah, plus that it currently seems that only the first call to `compileGlobal` for the same local returns `false` if compilation fails currently, while subsequent calls see that it is already `CommonFlags.COMPILED` and return `true`. I think we should fix that, instead of adding checks for `Type.void` that are only necessary due to a more general problem.

Reply:
Updated :+1: